File: BookLite/src/main/java/org/cubexmc/booklite/service/PdcKeys.java
package org.cubexmc.booklite.service;

import org.bukkit.NamespacedKey;
import org.cubexmc.booklite.BookLitePlugin;

public class PdcKeys {

    public static final int SCHEMA_VERSION = 1;

    private final NamespacedKey bookId;
    private final NamespacedKey generation;
    private final NamespacedKey version;

    public PdcKeys(BookLitePlugin plugin) {
        this.bookId = new NamespacedKey(plugin, "book_id");
        this.generation = new NamespacedKey(plugin, "generation");
        this.version = new NamespacedKey(plugin, "version");
    }

    public NamespacedKey bookId() { return bookId; }
    public NamespacedKey generation() { return generation; }
    public NamespacedKey version() { return version; }
}

File: BookLite/src/main/java/org/cubexmc/booklite/service/PdcKeys.kt
package org.cubexmc.booklite.service

import org.bukkit.NamespacedKey
import org.cubexmc.booklite.BookLitePlugin

class PdcKeys(plugin: BookLitePlugin) {
    private val bookId = NamespacedKey(plugin, "book_id")
    private val generation = NamespacedKey(plugin, "generation")
    private val version = NamespacedKey(plugin, "version")

    fun bookId(): NamespacedKey = bookId

    fun generation(): NamespacedKey = generation

    fun version(): NamespacedKey = version

    companion object {
        const val SCHEMA_VERSION: Int = 1
    }
}
